augur/align.py

```diff
@@ -51,79 +38,155 @@ def run(args):
+                raise AlignmentError("ERROR: Provided existing alignment ({}bp) is not the same length as the reference sequence ({}bp)".format(alignment.get_alignment_length(), len(ref_seq)))
+            existing_aln.append(ref_seq)
+            SeqIO.write(existing_aln, existing_aln_fname, 'fasta')
+            temp_files_to_remove.append(existing_aln_fname)
+            ref_name = ref_seq.id
+        else:
+            existing_aln_fname = args.existing_alignment # may be False
+
+        ## Given >=1 sequence files, create a single file for alignment. Add in the reference
+        ## sequence if desired
+        if args.reference_sequence and not existing_aln:
+            seqs_to_align_fname = args.output+".to_align.fasta"
+            temp_files_to_remove.append(seqs_to_align_fname)
+            ref_seq = read_reference(args.reference_sequence)
+            SeqIO.write(list(seqs.values())+[ref_seq], seqs_to_align_fname, 'fasta')
+            ref_name = ref_seq.id
+        elif len(args.sequences) > 1:
+            if existing_aln:
+                seqs_to_align_fname = args.output+".new_seqs_to_align.fasta"
             else:
-                ref_name = ref_seq.id
-                seq_fname+='.ref.fasta'
-                SeqIO.write(list(seqs.values())+[ref_seq], seq_fname, 'fasta')
+                seqs_to_align_fname = args.output+".to_align.fasta"
+            SeqIO.write(list(seqs.values()), seqs_to_align_fname, 'fasta')
+            temp_files_to_remove.append(seqs_to_align_fname)
         else:
-            print("WARNING: Cannot read reference sequence."
-                  "\n\tmake sure the file %s does not exist"%ref_fname)
+            seqs_to_align_fname = args.sequences[0]
+
+        check_duplicates(existing_aln, ref_name, seqs)
+
+        # before aligning, make a copy of the data that the aligner receives as input (very useful for debugging purposes)
+        if not existing_aln:
+            copyfile(seqs_to_align_fname, args.output+".pre_aligner.fasta")
+
+        # generate alignment command & run
+        cmd = generate_alignment_cmd(args.method, args.nthreads, existing_aln_fname, seqs_to_align_fname, args.output, args.output+".log")
+        success = run_shell_command(cmd)
+        if not success:
+            raise AlignmentError("Error during alignment")
+
+        # after aligning, make a copy of the data that the aligner produced (useful for debugging)
+        copyfile(args.output, args.output+".post_aligner.fasta")
+
+        # convert the aligner output to upper case (replacing the file in place)
+        write_uppercase_alignment_in_place(args.output)
+
+        # if we've specified a reference, strip out all the columns not present in the reference
+        # this will overwrite the alignment file
+        if ref_name:
+            seqs = strip_non_reference(args.output, ref_name, keep_reference=not args.remove_reference)
+            if args.fill_gaps:
```

Comment:
this might be useful even when running with an existing alignment. At least I don't see a strong reason not to allow it if people have to explicitly specify `fill_gaps`.

Reply:
Agreed, I think this was probably the original intention. Done in 05864fa